From 67d62906bfb5065a39ef060035a2d81bac24b707 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 10 Nov 2021 10:03:51 +0200
Subject: [PATCH] [SW-0] gaudi2: arc: update boot info base address

Currently for secureboot flow, bootinfo base is set to scratchpad SRAM
which causes failures to read bootinfo in zephyr and uboot as they
expect bootinfo to be present at 0x10000 offset of ARC0 ICCM.
This patch update bootinfo base to ARC0 ICCM offset 0x10000.

Change-Id: I953e1d72a0539d68b8cf4513c4bd78fa7b1f2fbc
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index c337dc5..7f2c80b 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -63,8 +63,8 @@
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + PPBOOT_START_OFFSET)
 #define PREBOOT_TEXT_BASE	(SDRAM_BASE + PREBOOT_START_OFFSET)
 
-#define SP_SRAM_BOOT_INFO_OFFSET	0x1800
-#define BOOT_INFO_BASE		(SP_SRAM_BASE + SP_SRAM_BOOT_INFO_OFFSET)
+#define ICCM_BOOT_INFO_OFFSET   0x10000
+#define BOOT_INFO_BASE          (ARC0_ICCM_BASE + ICCM_BOOT_INFO_OFFSET)
 
 /* In order not to conflict with ARC BOOTROM's usage of DCCM,
  * Let PPBoot start from the offset of 0x402C, for using DCCM
